llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -12179,6 +12217,8 @@ class BoUpSLP::ShuffleInstructionBuilder final : public BaseShuffleAnalysis {
         }
     }
     int VF = cast<FixedVectorType>(V1->getType())->getNumElements();
+    if (auto *VecTy = dyn_cast<FixedVectorType>(ScalarTy))
+      VF /= VecTy->getNumElements();
```

Comment:
Should this be processed by something similar like getVF or getVF itself?

Reply:
We can make `getVF` into a protected member function of `BaseShuffleAnalysis`. However, `BaseShuffleAnalysis` would need to store `ScalarTy`. See https://github.com/llvm/llvm-project/pull/99606/commits/f1de612dbf027f53e1f7dff1616aa1a98f2acd26.